#!/bin/bash
#PBS -S /bin/bash

## I want 4 processors 
#PBS -l procs=4

## How much RAM does each processor need?
#PBS -l pmem=2000mb

## The maximum walltime that will be used for my job 
#PBS -l walltime=04:15:00

## I want email sent when the job begins, ends and aborts (bea)
#PBS -m bea

## Where I want the email to be sent
#PBS -M YourEmailAddress@gmail.com 

## Let's run this job from the place that we submit the script from 
cd $PBS_O_WORKDIR

## Define variables to be used in the commands below
WORKING_DIR='/global/scratch/ARC_Training/Session3/RICHMOND/'
GENOME='/global/scratch/ARC_Training/Session3/Files/GENOME/genome.fa'
FASTQR1='/global/scratch/ARC_Training/Session3/Files/RAW_DATA/NA20845.chr19.subregion_R1.fastq'
FASTQR2='/global/scratch/ARC_Training/Session3/Files/RAW_DATA/NA20845.chr19.subregion_R2.fastq'
BAM='NA20845.chr19.subregion.bam'
SORTEDBAM='NA20845.chr19.subregion.sorted.bam'
SORTEDBAMPREFIX='NA20845.chr19.subregion.sorted'
SAM='NA20845.chr19.subregion.sam'


## Load the module containing bwa and samtools
module load bio-tools
module load bio-tools/WG_Genomic_Analysis_Workshop

## Let's go to directory we want to work in
cd $WORKING_DIR

## Generate BWA index (I already did this for you since it takes a long time)
## But if you want to know how to do it, it's simple:
#bwa index <in.fasta>
#bwa index $GENOME

## Map with BWA
bwa mem -t 4  $GENOME \
$FASTQR1 $FASTQR2 \
> $WORKING_DIR$SAM

## Convert sam to bam using samtools view
samtools view  -b $WORKING_DIR$SAM -o $WORKING_DIR$BAM 

## Sort the bam file using samtools sort
samtools sort $WORKING_DIR$BAM $WORKING_DIR$SORTEDBAMPREFIX 

## Index the sorted bam 
samtools index $WORKING_DIR$SORTEDBAM  


