#!/bin/bash
#PBS -S /bin/bash

## I want 4 processors 
#PBS -l procs=4

## How much RAM does each processor need?
#PBS -l pmem=2000mb

## The maximum walltime that will be used for my job 
#PBS -l walltime=200:15:00

## I want email sent when the job begins, ends and aborts (bea)
#PBS -m bea

## Where I want the email to be sent
#PBS -M phillip.a.richmond@gmail.com

## Let's run this job from the place that we submit the script from 
cd $PBS_O_WORKDIR

## Define variables to be used in the commands below
WORKING_DIR='/global/scratch/ARC_Training/Session6/Files/PROCESS/'
GENOME='/global/scratch/ARC_Training/Session6/Files/GENOME/mm10.fa'
GENOME_INDEX='/global/scratch/ARC_Training/Session6/Files/GENOME/mm10_hisat2_index'
SPLICE_SITES='/global/scratch/ARC_Training/Session6/Files/GENOME/mm10_splice_sites.txt'
GENOME_GTF='/global/scratch/ARC_Training/Session6/Files/GENOME/mm10_refseq.gtf'
FASTQR1='/global/scratch/ARC_Training/Session6/Files/RAW_DATA/ISS_S_2_chr13_R1.fastq'
FASTQR2='/global/scratch/ARC_Training/Session6/Files/RAW_DATA/ISS_S_2_chr13_R2.fastq'
SAM='ISS_S_2_hisat2.sam'
BAM='ISS_S_2_hisat2.bam'
SORTEDBAMPREFIX='ISS_S_2_hisat2.sorted'
SORTEDBAM='ISS_S_2_hisat2.sorted.bam'
SAMPLE_ID='ISS_S_2'
SAMPLE_GTF='ISS_S_2_hisat2_stringtie.gtf'
BALLGOWN_DIR='ISS_S_2_Ballgown/'
## Load the module containing bwa and samtools
module load bio-tools/WG_Genomic_Analysis_Workshop 

## Let's go to directory we want to work in
cd $WORKING_DIR

## Generate hisat2 index (I already did this for you since it takes a long time)
## But if you want to know how to do it, it's simple:
#hista2-build <in.fasta> <prefix> 
#hisat2-build $GENOME $GENOME_INDEX

## Map with hisat2
hisat2 -p 4 -x $GENOME_INDEX \
-1 $FASTQR1 -2 $FASTQR2 \
 --rna-strandness FR \
 --fr \
 --known-splicesite-infile $SPLICE_SITES \
 --rg-id "$SAMPLE_ID" \
 -S $WORKING_DIR$SAM

## Convert sam to bam using samtools view
samtools view -@ 4 -b $WORKING_DIR$SAM -o $WORKING_DIR$BAM 

## Sort the bam file using samtools sort
samtools sort -@ 4 $WORKING_DIR$BAM $WORKING_DIR$SORTEDBAMPREFIX

## Index the sorted bam 
samtools index $WORKING_DIR$SORTEDBAM  

## Use Stringtie to assemble transcripts
stringtie -G $GENOME_GTF \
--fr -o $WORKING_DIR$SAMPLE_GTF \
$WORKING_DIR$SORTEDBAM

## Use Stringtie to quantify against known transcripts
stringtie -e -b $WORKING_DIR$BALLGOWN_DIR \
 -G $GENOME_GTF --fr \
 -o $WORKING_DIR$SAMPLE_GTF \
 $WORKING_DIR$SORTEDBAM




